python/ray/experimental/ui.py

```diff
@@ -275,79 +277,162 @@ def task_timeline():
         description="View options:",
         disabled=False,
     )
+    obj_dep = widgets.Checkbox(
+        value=True,
+        description="Object dependencies",
+        disabled=False
+    )
+    task_dep = widgets.Checkbox(
+        value=True,
+        description="Task dependencies",
+        disabled=False
+    )
 
     start_box, end_box, range_slider, time_opt = get_sliders(False)
+    display(task_dep)
+    display(obj_dep)
     display(breakdown_opt)
     display(path_input)
 
-    def find_trace2html():
-        trace2html = "/tmp/ray/catapult/tracing/bin/trace2html"
+    def find_catapult():
         # Clone the catapult repository if it doesn't exist. TODO(rkn): We
         # could do this in the build.sh script later on.
-        if not os.path.exists(trace2html):
+        catapult_home = "/tmp/ray/catapult"
+        catapult_commit = "33a9271eb3cf5caf925293ec6a4b47c94f1ac968"
+        try:
+            # Check if we're inside a git repo
             cmd = ["git",
-                   "clone",
-                   "https://github.com/catapult-project/catapult.git",
-                   "/tmp/ray/catapult"]
-            subprocess.check_output(cmd)
-            print("Cloning catapult to /tmp/ray/catapult.")
-        assert os.path.exists(trace2html)
-        return trace2html
+                   "-C",
+                   catapult_home,
+                   "rev-parse",
+                   "--is-inside-work-tree"]
+            subprocess.check_call(cmd)
+
+            return catapult_home
+        except subprocess.CalledProcessError:
+            # Error on non-zero exit code (e.g. - ".git not found")
+            pass
+
+        print("Cloning catapult to {}.".format(catapult_home))
+        cmd = ["git",
+               "clone",
+               "https://github.com/catapult-project/catapult.git",
+               catapult_home]
+        subprocess.check_call(cmd)
+
+        # Checks out the commit associated with allowing different arrow
+        # colors. This can and should be removed after catapult's next
+        # release.
+        print("Checking out commit {}.".format(catapult_commit))
+        cmd = ["git", "-C", catapult_home, "checkout", catapult_commit]
+        subprocess.check_call(cmd)
+
+        return catapult_home
 
     def handle_submit(sender):
         tmp = tempfile.mktemp() + ".json"
```

Comment:
This handler is doing a lot of things, I think all of the logic of cloning catapult and autogenerating files should be separated into a helper function (and the helper function should have a docstring that says precisely what it does)